core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -734,9 +734,20 @@ public BytesReference ext() {
         return ext;
     }
 
+    /**
+     * Create a new SearchSourceBuilder with attributes set by an xContent.
+     */
     public SearchSourceBuilder fromXContent(XContentParser parser, QueryParseContext context, AggregatorParsers aggParsers)
             throws IOException {
         SearchSourceBuilder builder = new SearchSourceBuilder();
+        builder.parseXContent(parser, context, aggParsers);
```

Comment:
I don;t understand why this needed to change? Surely the parser for the reindex API could just call fromXContent as it was when it came across the object representing the search body?